cypress-tests/cypress/support/redirectionHandler.js

```diff
@@ -399,9 +413,15 @@ function verifyReturnUrl(redirection_url, expected_url, forward_flow) {
   const paymentStatus = urlParams.get('status');
 
   // Check for valid statuses
-  if (!paymentStatus) {
-    console.error(`Error: Payment status is undefined. Possible 5xx or 4xx error during redirection.`);
-    throw new Error(`Payment status is undefined. Possible 5xx or 4xx error during redirection.`);
+  if (statusCode >= 400 && statusCode < 500) {
+    console.error(`Client Error: ${statusCode} - ${paymentStatus || "Payment status not available"}`);
+    throw new ClientError(`Client error occurred. Status code: ${statusCode}`);
+  } else if (statusCode >= 500) {
+    console.error(`Server Error: ${statusCode} - ${paymentStatus || "Payment status not available"}`);
+    throw new ServerError(`Server error occurred. Status code: ${statusCode}`);
+  } else if (!paymentStatus) {
+    console.error(`Error: Payment status is undefined. Unexpected error.`);
+    throw new Error(`Payment status is undefined. Unexpected error.`);
```

Comment:
From what I understand, this would not work as expected.  
The reason being, you will get a 2xx response with a `url` in `next_action` and the `paymentStatus` would be `requires_customer_action`.  
The possibility that @Gnanasundari24 mentioned was about getting a `4xx` during redirection. You would not get a status_code over there. For example:

<img width="1678" alt="image" src="https://github.com/user-attachments/assets/71482784-31a8-40f6-a710-98b178b7f4f1">

But you can definitely see the `4xx` when you open inspect tab.

So, you would have either have to throw an error directly like what is done in commit [`e1f1e44` (#6187)](https://github.com/juspay/hyperswitch/pull/6187/commits/e1f1e44e2a93f0b33e198443e1fd55eadbb537a5) or read what is mentioned in the screen which I do not think is possible in a line or 2.

Reply:
Check if latest commit resolve it